pandas/core/indexes/category.py

```diff
@@ -309,14 +347,44 @@ def tolist(self):
 
     @property
     def codes(self):
+        """
+        Return codes attribute of self._data.
```

Comment:
We should not refer to `self._data` as this is a private implementation detail the user should not be aware of.
 
But you can use the explanation you put above in the CategoricalIndex docstring (the part I asked you to remove again)

And similar comment for `categories` and `ordered` below